app/domain/authentication/authn_jwt/orchestrate_authentication.rb

```diff
@@ -0,0 +1,47 @@
+require 'command_class'
+
+# This class is the starting point of the JWT authenticate requests, responsible to identify the vendor configuration and to run the JWT authenticator
+module Authentication
+  module AuthnJwt
+
+    OrchestrateAuthentication ||= CommandClass.new(
+      dependencies: {
+        validate_uri_based_parameters: Authentication::AuthnJwt::InputValidation::ValidateUriBasedParameters.new,
+        configuration_factory: Authentication::AuthnJwt::VendorConfigurations::ConfigurationFactory.new,
+        jwt_authenticator: Authentication::AuthnJwt::Authenticator.new,
+        logger: Rails.logger,
+        enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str
```

Comment:
does this need to be an input? i think that this will not work if someone enables an authenticator in the DB

Reply:
@sashaCher can you take this comment?

Reply:
Enabled authenticators contain db enabled authenticators https://github.com/cyberark/conjur/blob/a48be7298d583c86bbe22817993b221f9f48b4b2/app/domain/authentication/installed_authenticators.rb#L36

Reply:
yes but if we set the `enabled_authenticators` object as a dependency then it is created only once, at the server startup. so if the db is changed in run-time we won't get this change.

I suspect that this is the case, didn't verify it.

Reply:
It looks like the code flow behind `enabled_authenticators_str` is stateless means there're no "memorisation" `||` statements and no class member access `@` operations...

Reply:
yes but when a command class is created then it creates its dependencies. whereas inputs are created in run-time. if you wish to leave the `enabled_authenticators` as a dependency please verify that it works.

Reply:
Fixed good catch